iocage/lib/ZFS.py

```diff
@@ -24,16 +24,31 @@
 """iocage libzfs enhancement module."""
 import typing
 import libzfs
+import datetime
 
 import iocage.lib.Logger
 import iocage.lib.helpers
 import iocage.lib.errors
 
+Logger = iocage.lib.Logger.Logger
+
 
 class ZFS(libzfs.ZFS):
     """libzfs enhancement module."""
 
-    logger: typing.Optional[iocage.lib.Logger.Logger] = None
+    _logger: typing.Optional['iocage.lib.Logger.Logger']
```

Comment:
This change is required for compliance with the typing. The logger on ZFS is optional and is not set when initializing the object, but can be set at any later time (using the setter method).

Reply:
i think it might be time that we (once again) rethink our logger (initialization (method))

Reply:
This is a special occasion. Having a logger in this class available improves our output, but the ZFS class itself should not depend on having a logger. (Imagine you want to use the module in another context than libiocage). I also remember an issue overriding the default ZFS __init__ method.